link/link.go

```diff
@@ -49,6 +50,8 @@ type RawLinkOptions struct {
 	Program *ebpf.Program
 	// Attach must match the attach type of Program.
 	Attach ebpf.AttachType
+	// BTF is the BTF of the attachment target (or 0).
+	BTF btf.Type
```

Comment:
Why make this a `btf.TypeID` instead of a `btf.Type`? Other fields here use the Go native type instead of the ID, such as `Program`.
(The comment about it being 0 is incorrect based on current implementation. It should have said nil, but I think it also doesn't need to be said.)

Reply:
My thinking was this (if I remember correctly): most of the code we have constructs a `*Func` to pass here, only for the Go type to be discarded. To look at it another way: since this is a low-level interface (hence the Raw in the name) it's better to stay close to the kernel interface.